libc/string/string_tokpbrkstr.c

```diff
@@ -0,0 +1,687 @@
+
+	free(extAsciiStr);
+	extAsciiStr = create_extAscii_set();
+	/*
+	 * Checking available tokenization on all elements
+	 * from outside of ASCII charset one by one, with
+	 * the first separator set as the first element of the set
+	 */
+	TEST_ASSERT_EQUAL_PTR(extAsciiStr, strtok_r(extAsciiStr, "\128", &rest));
+	for (i = ASCII_LENGTH + 1; i < sizeof(separator); i++) {
+		separator[i - ASCII_LENGTH] = i;
+		TEST_ASSERT_EQUAL_PTR(&extAsciiStr[i - ASCII_LENGTH], strtok_r(NULL, &separator[i - ASCII_LENGTH], &rest));
+	}
+
+	free(extAsciiStr);
+#endif
+}
+
+
+TEST(string_tok_r, restriction)
+{
+#ifdef __phoenix__
+	TEST_IGNORE();
+#else
+
+	char *rest[ASCII_LENGTH] = { 0 },
+		 *tok,
+		 *asciiStr = testdata_createCharStr(ASCII_LENGTH);
+
+	int i;
+
+	for (i = 1; i < ASCII_LENGTH - 1; i++) {
+		TEST_ASSERT_EQUAL_PTR(&asciiStr[1], tok = strtok_r(&asciiStr[1], &asciiStr[i + 1], &rest[i]));
+		TEST_ASSERT_EQUAL_CHAR_ARRAY(&asciiStr[1], tok, i);
+		TEST_ASSERT_NOT_NULL(rest[i]);
+	}
+
+	free(asciiStr);
+#endif
+}
+
+
+/*
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_str)
+{
+}
+
+
+TEST_TEAR_DOWN(string_str)
+{
+}
+
+
+TEST(string_str, basic)
+{
+	char loremIpsum[BUFFSIZE] = LOREM_IPSUM,
+		 *asciiStr = testdata_createCharStr(ASCII_LENGTH);
+
+	/* Standard use of strstr on arrays */
+	TEST_ASSERT_EQUAL_PTR(loremIpsum, strstr(loremIpsum, "Lorem"));
+	TEST_ASSERT_EQUAL_PTR(&loremIpsum[6], strstr(loremIpsum, "Ipsum"));
+	TEST_ASSERT_EQUAL_PTR(&loremIpsum[12], strstr(loremIpsum, "Dolor"));
+
+	/* Simple usage of strstr with ascii set */
+	TEST_ASSERT_EQUAL_PTR(asciiStr, strstr(asciiStr, asciiStr));
+	TEST_ASSERT_EQUAL_STRING(&asciiStr[ASCII_LENGTH / 4], strstr(asciiStr, &asciiStr[ASCII_LENGTH / 4]));
+	TEST_ASSERT_EQUAL_STRING(&asciiStr[ASCII_LENGTH / 3], strstr(asciiStr, &asciiStr[ASCII_LENGTH / 3]));
+	TEST_ASSERT_EQUAL_STRING(&asciiStr[ASCII_LENGTH / 2], strstr(asciiStr, &asciiStr[ASCII_LENGTH / 2]));
+
+	/* Passing an empty string as the second parameter to get first */
+	TEST_ASSERT_EQUAL_STRING(&asciiStr[ASCII_LENGTH / 2], strstr(&asciiStr[ASCII_LENGTH / 2], ""));
+
+	free(asciiStr);
+}
+
+
+TEST(string_str, empty_args)
+{
+	/* Different scenarios of using empty input or output*/
+	TEST_ASSERT_NOT_NULL(strstr("", ""));
+	TEST_ASSERT_NOT_NULL(strstr("abc", "\0"));
+
+	TEST_ASSERT_NULL(strstr("", "abc"));
+	TEST_ASSERT_NULL(strstr("\0", "abc"));
+	TEST_ASSERT_NULL(strstr("abc", "def"));
+}
+
+
+TEST(string_str, out_of_ascii)
+{
+	int i;
+
+	char *extAsciiStr = create_extAscii_set();
+
+	for (i = 1; i < sizeof(extAsciiStr); i++) {
+		TEST_ASSERT_EQUAL_PTR(&extAsciiStr[i], strstr(extAsciiStr, &extAsciiStr[i]));
+	}
+
+	free(extAsciiStr);
+}
+
+
+/*
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_pbrk)
+{
+}
+
+
+TEST_TEAR_DOWN(string_pbrk)
+{
+}
+
+
+TEST(string_pbrk, basic)
```

Comment:
You can certainly add more cases to cover this funciton. First example would be strbrk("abc", "") For example:

- passing bytes in the `s2` in various order, so strpbrk("abcde", "dce")
- passing more bytes (than available in `s1`) in `s2`, so sth like: strpbrk("abcde", "cdefgh")
- big buffers
- a few same bytes in s2
- maybe some other cases

Reply:
I tried different approaches in those cases, I will wait for a response about them.